server/eclipse-project/src/main/java/us/freeandfair/corla/json/UploadedFileJsonAdapter.java

```diff
@@ -0,0 +1,138 @@
+ * @author Daniel M. Zimmerman <dmz@freeandfair.us>
+ * @description A system to assist in conducting statewide risk-limiting audits.
+ */
+
+package us.freeandfair.corla.json;
+
+import java.io.IOException;
+
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import us.freeandfair.corla.model.UploadedFile;
+import us.freeandfair.corla.persistence.Persistence;
+
+/**
+ * JSON adapter for uploaded files. This enapsulates in JSON all the information
+ * about the file except for its actual contents, and retrieves from the database
+ * the uploaded file corresponding to the specified county and database ID 
+ * ignoring other fields; note that "null" can be retrieved if there is no
+ * matching uploaded file in the database.
+ * 
+ * @author Daniel M. Zimmerman
+ * @version 0.0.1
+ */
+// the default constructor suffices for type adapters
+@SuppressWarnings("PMD.AtLeastOneConstructor")
+public final class UploadedFileJsonAdapter 
+    extends TypeAdapter<UploadedFile> {
+  /**
+   * The "file_id" string (for JSON serialization).
+   */
+  private static final String FILE_ID = "file_id";
+  
+  /**
+   * The "county_id" string (for JSON serialization).
+   */
+  private static final String COUNTY_ID = "county_id";
+  
+  /**
+   * The "filename" string (for JSON serialization).
+   */
+  private static final String FILENAME = "filename";
+  
+  /**
+   * The "size" string (for JSON serialization).
+   */
+  private static final String SIZE = "size";
+
+  /**
+   * The "timestamp" string (for JSON serialization).
+   */
+  private static final String TIMESTAMP = "timestamp";
+  
+  /**
+   * The "hash_status" string (for JSON serialization).
+   */
+  private static final String HASH_STATUS = "hash_status";
+  
+  /**
+   * The "status" string (for JSON serialization).
+   */
+  private static final String STATUS = "status";
+  
+  /**
+   * Writes an uploaded file object.
+   * 
+   * @param the_writer The JSON writer.
+   * @param the_file The object to write.
+   */ 
+  @Override
+  public void write(final JsonWriter the_writer, 
+                    final UploadedFile the_file) 
+      throws IOException {
+    the_writer.beginObject();
+    the_writer.name(FILE_ID).value(the_file.id());
+    the_writer.name(COUNTY_ID).value(the_file.countyID());
+    the_writer.name(FILENAME).value(the_file.filename());
+    the_writer.name(SIZE).value(the_file.size());
+    the_writer.name(TIMESTAMP).value(the_file.timestamp().toString());
+    the_writer.name(HASH_STATUS).value(the_file.hashStatus().toString());
+    the_writer.name(STATUS).value(the_file.status().toString());
+    the_writer.endObject();
+  }
+  
+  /**
+   * Reads an uploaded file object.
+   * 
+   * @param the_reader The JSON reader.
+   * @return the object.
+   */
+  @Override
+  public UploadedFile read(final JsonReader the_reader) 
```

Comment:
What's going on with this method? Did you mean to start implementing this?

Reply:
I find myself wondering if I failed to push a commit to this branch.

Reply:
Um... no, the whole method is there. What's going on with your comment? :)

Reply:
I thought maybe you hadn't finished implementing this. For example:
- Switch cases without breaks and what seem to be fallthroughs that we don't want. Are you just batching okay-to-ignore keys with a fallthrough case?
- Specifically, I was confused by skipping the filename field if we have one.

Reply:
Looks fine to me.

Reply:
If you're wondering why everything other than file ID is ignored, the answer is, because it's irrelevant to getting the file with that ID out of the database and we're checking that the correct county owns that file before importing it anyway.

Reply:
(i.e., it's to make the client's life potentially easier by having to submit a smaller file descriptor object, namely `{ "id": blah }`), but also being able to just submit the reply verbatim.

Reply:
@dmzimmerman that explanation makes sense.